### PR TITLE
VCDA-197 : [Independent Disks] Update an existing independent disk size,iops and storage profile

### PR DESCRIPTION
**Pull request by sompa**

### CODE REVIEW

File: pyvcloud/vcd/vdc.py [hunk 1]
```diff
@@ -415,6 +426,14 @@ def update_disk(self, name, size, new_name=None, description=None,
         else:
             disk = self.get_disk(name)
 
+        if storage_profile_name is not None:
+            storage_profile = self.get_storage_profile(storage_profile_name)
+            sp = etree.XML(etree.tostring(storage_profile, pretty_print=True))
+            sp_href = sp.attrib['href']
+            diskParms.append(E.StorageProfile(href=sp_href,
+                                              name=storage_profile_name))
+            print(etree.tostring(diskParms, pretty_print=True))
```
Comment (anusuyar) [inline pyvcloud/vcd/vdc.py]: Do we need to print the diskParams here? Will it be seen by the vcd-cli user? If so, does he need to see the diskparams?

**Review by anusuyar — APPROVED** [review]

File: tests/vcd_disk.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def test_008_update_disk(self):
         vdc = VDC(self.client, href=v.get('href'))
 
         
-        taskObj = vdc.update_disk(self.config['vcd']['disk'], "20", "DiskDisk1a", "20 MB Disk")
+        taskObj = vdc.update_disk(self.config['vcd']['disk'], "3072", "DiskUpdated", "3 GB Disk")
```
Comment (rocknes) [inline tests/vcd_disk.py:74]: Why is this change necessary?

File: tests/vcd_disk.py [hunk 3]
```diff
@@ -87,8 +87,54 @@ def test_008_update_disk(self):
...
- 
 
+    def test_008_update_disk_iops(self):
+        logged_in_org = self.client.get_org()
+
+        org = Org(self.client, resource=logged_in_org)
+        v = org.get_vdc(self.config['vcd']['vdc'])
+        vdc = VDC(self.client, href=v.get('href'))
+
+        taskObj = vdc.update_disk(self.config['vcd']['disk'], "3072", "DiskUpdated", "3 GB Disk", "stf-storage-service-gold", "200")
```
Comment (rocknes) [inline tests/vcd_disk.py]: Too long line

File: tests/vcd_disk.py [hunk 4]
```diff
@@ -87,8 +87,54 @@ def test_008_update_disk(self):
         assert task.get('status') == TaskStatus.SUCCESS.value
 
         print ("Updating VDC disk: " + str(etree.tostring(task, pretty_print=True), "utf-8"))
- 
 
+    def test_008_update_disk_iops(self):
```
Comment (rocknes) [inline tests/vcd_disk.py:91]: change the test number to something unique

**Review by rocknes — APPROVED** [review]

File: tests/vcd_disk.py (hunk 2 above)
Comment (sompa) [reply tests/vcd_disk.py:74]: Wanted to test out with my own set of inputs.

File: pyvcloud/vcd/vdc.py (hunk 1 above)
Comment (sompa) [reply pyvcloud/vcd/vdc.py]: Its not, i will remove it.

File: tests/vcd_disk.py (hunk 3 above)
Comment (sompa) [reply tests/vcd_disk.py]: changed.

File: tests/vcd_disk.py (hunk 4 above)
Comment (sompa) [reply tests/vcd_disk.py:91]: How does the test number matter here ? The test names are unique already.

**Comment by sompa** [comment]
Closing this review and pull request and creating one directly on vmware/pyvcloud.
